### PR TITLE
Transformations: Window calculation should use 0 if value in window does not exist

### PR DESCRIPTION
**Pull request by gelicia**
**What is this feature?**

This updates the window calculation such that if a value does not exist, it counts as 0.

This is somewhat hard to replicate because in some scenarios if the value doesnt exist, it will be null, and in other scenarios it will be undefined. Javascript can `-= null` and have null count as 0, but with `-= undefined`, it will return NaN. With this logic, if it is either null or undefined it will count as 0.

**Why do we need this feature?**

So joined queries where data does not exist in all time fields will calculate properly without returning NaN.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/support-escalations/issues/13051

**Special notes for your reviewer:**
I initially tried reproing this with a CSV dataset, either 3 queries or 1 query with some empty values and neither of them worked. 3 queries does not work and 1 query with some empty values returns null instead of undefined.

I ended up importing data into a postgres database I have running locally and having 3 queries coming from one table with different WHERE clauses. Then you can see it. I dont want to export out the whole panel because the data set was derived from customer data. Let me know if you would like me to set up something for reproing this.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: packages/grafana-data/src/transformations/transformers/calculateField.ts [hunk 1]
```diff
@@ -357,7 +357,7 @@ function getTrailingWindowValues(frame: DataFrame, reducer: ReducerID, selectedF
         sum += currentValue;
 
         if (i > window - 1) {
-          sum -= selectedField.values[i - window];
+          sum -= selectedField.values[i - window] != null ? selectedField.values[i - window] : 0;
```
Comment (leeoniya) [inline packages/grafana-data/src/transformations/transformers/calculateField.ts:360]: maybe do `sum -= (val ?? 0)`

File: packages/grafana-data/src/transformations/transformers/calculateField.test.ts [hunk 2]
```diff
@@ -728,6 +728,39 @@ describe('calculateField transformer w/ timeseries', () => {
...
+    const series = toDataFrame({
+      fields: [{ name: 'x', type: FieldType.number, values: [1, undefined, 3, 4, 5] }],
+    });
+
+    await expect(transformDataFrame([cfg], [series])).toEmitValuesWith((received) => {
+      const data = received[0][0];
+
+      //console.log(data.fields);
+
+      expect(data.fields.length).toEqual(2);
```
Comment (gelicia) [inline packages/grafana-data/src/transformations/transformers/calculateField.test.ts:755]: @leeoniya I feel like this test is right but the results are wrong. Should it be 1, 0.5, 1.5, 3.5, 4.5 ?

Comment (leeoniya) [reply packages/grafana-data/src/transformations/transformers/calculateField.test.ts:755]: yeah, agreed. these results look like a rolling sum, not mean.

Comment (gelicia) [reply packages/grafana-data/src/transformations/transformers/calculateField.test.ts:755]: Good news: I fixed it by ensuring that the current value the window lands on also evaluates to 0 if it is null. I then get what I'd expect.

Bad news: this did end up changing the results of another test. I do agree with the change, but I could see it being problematic if the previous results were evaluated to be correct at the time. Not sure how to proceed to be honest.

**Comment by gelicia** [comment]
Talked with Leon - decided this logic is flawed. Rather than have calculation be 0 if the value does not exist, we should not count it and only do means with the number of values that do exist.

**Review by leeoniya — APPROVED** [review]

**Pull request merged**